solutions/usaco-988.mdx

```diff
@@ -7,14 +7,81 @@ author: Ananth Kashyap
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_photo_bronze_jan20.html)
 
+Since $N \leq 1000$, we can simulate every ordering and pick the first valid permutation (since that would be the lexicographical minimum).
+
 ## Implementation
 
-**Time Complexity:** $\mathcal{O}(N^3)$
 
 <LanguageSection>
 
+<CPPSection>
+
+**Time Complexity:** $\mathcal{O}(N^2)$
+
+```cpp 
+#include <iostream>
+#include <vector>
+#include <algorithm>
```

Comment:
```suggestion
#include <algorithm>
#include <cstdio>
```
maybe it compiles fine without it on some machines, but [freopen is defined in cstdio](https://en.cppreference.com/w/cpp/io/c/freopen) so i think it would be better to include it